extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/OnnxModelRunner.java

```diff
@@ -0,0 +1,26 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor;
+
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OrtEnvironment;
+import ai.onnxruntime.OrtException;
+import ai.onnxruntime.OrtSession;
+
+import java.util.Collections;
+
+public class OnnxModelRunner {
+
+    private OrtSession session;
+
+    private OrtEnvironment environment;
+
+    public OnnxModelRunner(byte[] onnxModelBytes) throws OrtException {
+        environment = OrtEnvironment.getEnvironment();
+        final OrtSession.SessionOptions options = new OrtSession.SessionOptions();
+        session = environment.createSession(onnxModelBytes, options);
+    }
+
+    public OrtSession.Result runModel(String[][] throttlingInferenceRow) throws OrtException {
+        final OnnxTensor inputTensor = OnnxTensor.createTensor(OrtEnvironment.getEnvironment(), throttlingInferenceRow);
+        return session.run(Collections.singletonMap("input", inputTensor));
```

Comment:
is it a long-running operation? Should we consider calling it in a separate thread performance-wise?

Reply:
In terms of prediction time
- 1st warmup run after the model is fetched and put in cache: 8.5 ms
- all the following runs: 0.4 - 0.5 ms

The runs are tests in the bid request with 4 imps and 5 bidders per imp

[01_rtd_bid_request_test_banner_native.json](https://github.com/user-attachments/files/16839764/01_rtd_bid_request_test_banner_native.json)